### PR TITLE
ROX-9094: Create tasks for failing tests

### PR DESCRIPTION
**Pull request by janisz**
To collect more information about common test failures lets collect them in JIRA tickets. 
`junit2jira` is a small binary the investigate given path and searches for junit `xml` reports.
It creates issue for every failed test or if issue already exists it adds a comment with all details. 
This is a prototype of integration as proper integration will be implemented after migrating to GHA.

### CODE REVIEW

File: scripts/ci/lib.sh [hunk 1]
```diff
@@ -1185,6 +1185,13 @@ store_test_results() {
     local from="$1"
     local to="$2"
 
+    if ! is_in_PR_context; then
+        info "Creating JIRA task for failures found in $from"
+        wget -q https://github.com/janisz/junit2jira/releases/download/v0.0.1/junit2jira && \
+        chmod +x junit2jira && \
+        ./junit2jira -junit-reports-dir "$from"
```
Comment (gavin-stackrox) [inline scripts/ci/lib.sh:1192]: Given that it is a prototype have it silently fail and not disrupt following functionality:
```suggestion
{
        info "Creating JIRA task for failures found in $from"
        wget -q https://github.com/janisz/junit2jira/releases/download/v0.0.1/junit2jira && \
        chmod +x junit2jira && \
        ./junit2jira -junit-reports-dir "$from"
} || true
```

File: scripts/ci/lib.sh [hunk 2]
```diff
@@ -1185,6 +1185,13 @@ store_test_results() {
     local from="$1"
     local to="$2"
 
+    if ! is_in_PR_context; then
+        info "Creating JIRA task for failures found in $from"
+        wget -q https://github.com/janisz/junit2jira/releases/download/v0.0.1/junit2jira && \
```
Comment (gavin-stackrox) [inline scripts/ci/lib.sh:1190]:
```suggestion
        wget -q https://github.com/stackrox/junit2jira/releases/download/v0.0.1/junit2jira && \
```

**Review by gavin-stackrox — CHANGES_REQUESTED** [review]

File: scripts/ci/lib.sh (hunk 2 above)
Comment (janisz) [reply scripts/ci/lib.sh:1190]: https://github.com/stackrox/junit2jira/releases/tag/v0.0.1

**Review by porridge — APPROVED** [review]
I'd feel much safer if this ran *after* the `cp` in case `junit2jira` blows through some deadline or memory limit.

File: scripts/ci/lib.sh [hunk 3]
```diff
@@ -1185,6 +1185,15 @@ store_test_results() {
     local from="$1"
     local to="$2"
 
+    if ! is_in_PR_context; then
+    {
+        info "Creating JIRA task for failures found in $from"
+        wget -q https://github.com/stackrox/junit2jira/releases/download/v0.0.1/junit2jira && \
```
[the remark this answers is not in the document]
Comment (janisz) [reply scripts/ci/lib.sh:1191]: For some reason `curl` could not download the file but `wget` can.
wget has `--tries` option that by default is set to 20.

**Comment by janisz** [comment]
@gavin-stackrox PTAL

File: scripts/ci/lib.sh (hunk 3 above)
[the remark this answers is not in the document]
Comment (porridge) [reply scripts/ci/lib.sh:1191]: Interesting, works for me.
```
$ curl -o j2j -L https://github.com/stackrox/junit2jira/releases/download/v0.0.1/junit2jira
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 2317k  100 2317k    0     0   612k      0  0:00:03  0:00:03 --:--:--  869k
$ file j2j
j2j: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), statically linked, no section header
```

But I'm not insisting, 20 retries SGMT

[the remark this answers is not in the document]
Comment (janisz) [reply scripts/ci/lib.sh:1191]: My bad, I used `-l` instead of `-L`

**Review by gavin-stackrox — APPROVED** [review]
LGTM!

**Pull request merged**